applications/nrf5340_audio/src/audio/le_audio_rx.c

```diff
@@ -4,47 +4,42 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
+#include "le_audio_rx.h"
+
 #include <zephyr/kernel.h>
 #include <nrfx_clock.h>
+#include <zephyr/net_buf.h>
 
 #include "streamctrl.h"
 #include "audio_datapath.h"
 #include "macros_common.h"
 #include "audio_system.h"
 #include "audio_sync_timer.h"
+#include "audio_defines.h"
 
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(le_audio_rx, CONFIG_LE_AUDIO_RX_LOG_LEVEL);
 
-struct ble_iso_data {
-	uint8_t data[CONFIG_BT_ISO_RX_MTU];
-	size_t data_size;
-	bool bad_frame;
-	uint32_t sdu_ref;
-	uint32_t recv_frame_ts;
-} __packed;
-
 struct rx_stats {
 	uint32_t recv_cnt;
 	uint32_t bad_frame_cnt;
-	uint32_t data_size_mismatch_cnt;
 };
 
 static bool initialized;
 static struct k_thread audio_datapath_thread_data;
 static k_tid_t audio_datapath_thread_id;
 K_THREAD_STACK_DEFINE(audio_datapath_thread_stack, CONFIG_AUDIO_DATAPATH_STACK_SIZE);
 
-DATA_FIFO_DEFINE(ble_fifo_rx, CONFIG_BUF_BLE_RX_PACKET_NUM, WB_UP(sizeof(struct ble_iso_data)));
+DATA_FIFO_DEFINE(ble_fifo_rx, CONFIG_BUF_BLE_RX_PACKET_NUM, WB_UP(sizeof(struct audio_data)));
+NET_BUF_POOL_FIXED_DEFINE(ble_rx_pool, CONFIG_BUF_BLE_RX_PACKET_NUM, CONFIG_BT_ISO_RX_MTU, 0, NULL);
```

Comment:
`ble_rx_pool` could be moved to `le_audio_rx_data_handler` if you want to reduce number of global variables